src/probnum/diffeq/odesolution.py

```diff
@@ -1,176 +1,70 @@
-    >>> # Individual entries of the discrete-time solution can be accessed with
-    >>> print(solution[5])
-    <Normal with shape=(1,), dtype=float64>
-    >>> print(solution[5].mean)
-    [0.55945475]
-    >>> # Evaluate the continuous-time solution at a new time point t=0.65
-    >>> print(solution(0.65).mean)
-    [0.69875089]
-    """
+import probnum._randomvariablelist as pnrv_list
+import probnum.random_variables as pnrv
+import probnum.type
 
-    def __init__(self, times, rvs, solver):
 
-        # try-except is a hotfix for now:
-        # future PR is to move KalmanPosterior-info out of here, e.g. into GaussianIVPFilter
-        try:
-            self._kalman_posterior = KalmanPosterior(
-                times, rvs, solver.gfilt, solver.with_smoothing
-            )
-            self._t = None
-            self._y = None
-        except AttributeError:
-            self._kalman_posterior = None
-            self._t = times
-            self._y = _RandomVariableList(rvs)
-        self._solver = solver
+class ODESolution(abc.ABC):
+    """ODE Solution interface."""
 
     @property
-    def t(self):
-        """:obj:`np.ndarray`: Times of the discrete-time solution"""
-        if self._t:  # hotfix
-            return self._t
-        else:
-            return self._kalman_posterior.locations
+    @abc.abstractmethod
+    def t(self) -> np.ndarray:
+        """Time points of the discrete-time solution."""
+        raise NotImplementedError
 
     @property
-    def y(self):
-        """:obj:`list` of :obj:`RandomVariable`: Probabilistic discrete-time solution
-
-        Probabilistic discrete-time solution at times :math:`t_1, ..., t_N`,
-        as a list of random variables.
-        To return means and covariances use ``y.mean`` and ``y.cov``.
-        """
-        if self._y:  # hotfix
-            return self._y
-        else:
-            projmat = self._solver.prior.proj2coord(coord=0)
-            function_rvs = [projmat @ rv for rv in self._state_rvs]
-            return _RandomVariableList(function_rvs)
+    @abc.abstractmethod
+    def y(self) -> pnrv_list._RandomVariableList:
+        """Discrete-time solution."""
+        raise NotImplementedError
 
     @property
-    def dy(self):
-        """:obj:`list` of :obj:`RandomVariable`: Derivatives of the discrete-time solution"""
-        projmat = self._solver.prior.proj2coord(coord=1)
-        dy_rvs = [projmat @ rv for rv in self._state_rvs]
-        return _RandomVariableList(dy_rvs)
-
-    @property
-    def _state_rvs(self):
-        """:obj:`list` of :obj:`RandomVariable`:"""
-        return self._kalman_posterior.state_rvs
-
-    def __call__(self, t):
+    def dy(self) -> pnrv_list._RandomVariableList:
+        """First derivative of the discrete-time solution."""
+        raise NotImplementedError("The first derivative has not been implemented")
+
+    # Not abstract, because providing interpolation could sometimes be tedious.
```

Comment:
this comment. 

I think it can be annoying to enforce dense output. I find it least stressful to only enforce lists of random variables.

Reply:
I agree, I think we should not enforce general ODE solutions to provide dense output.